initSidebarItems({"enum":[["Error",""],["Scope","Identifies the an OAuth2 authorization scope. A scope is needed when requesting an authorization token."]],"struct":[["AlignReadGroupSetsRequest","The read group set align request."],["AlignReadGroupSetsResponse","The read group set align response."],["Annotation","An annotation describes a region of reference genome. The value of an annotation may be one of several canonical types, supplemented by arbitrary info tags. A variant annotation is represented by one or more of these canonical types. An annotation is not inherently associated with a specific sample or individual (though a client could choose to use annotations in this way). Example canonical annotation types are 'Gene' and 'Variant'."],["AnnotationBatchCreateCall","Creates one or more new annotations atomically. All annotations must belong to the same annotation set. Caller must have WRITE permission for this annotation set. For optimal performance, batch positionally adjacent annotations together."],["AnnotationCreateCall","Creates a new annotation. Caller must have WRITE permission for the associated annotation set."],["AnnotationDeleteCall","Deletes an annotation. Caller must have WRITE permission for the associated annotation set."],["AnnotationGetCall","Gets an annotation. Caller must have READ permission for the associated annotation set."],["AnnotationMethods","A builder providing access to all methods supported on *annotation* resources. It is not used directly, but through the `Genomics` hub."],["AnnotationPatchCall","Updates an annotation. The update must respect all mutability restrictions and other invariants described on the annotation resource. Caller must have WRITE permission for the associated dataset. This method supports patch semantics."],["AnnotationSearchCall","Searches for annotations that match the given criteria. Results are returned ordered by start position. Annotations that have matching start positions are ordered deterministically. Caller must have READ permission for the queried annotation sets."],["AnnotationSet","An annotation set is a logical grouping of annotations that share consistent type information and provenance. Examples of annotation sets include 'all genes from refseq', and 'all variant annotations from ClinVar'."],["AnnotationSetCreateCall","Creates a new annotation set. Caller must have WRITE permission for the associated dataset."],["AnnotationSetDeleteCall","Deletes an annotation set. Caller must have WRITE permission for the associated annotation set."],["AnnotationSetGetCall","Gets an annotation set. Caller must have READ permission for the associated dataset."],["AnnotationSetMethods","A builder providing access to all methods supported on *annotationSet* resources. It is not used directly, but through the `Genomics` hub."],["AnnotationSetPatchCall","Updates an annotation set. The update must respect all mutability restrictions and other invariants described on the annotation set resource. Caller must have WRITE permission for the associated dataset. This method supports patch semantics."],["AnnotationSetSearchCall","Searches for annotation sets that match the given criteria. Results are returned in a deterministic order. Caller must have READ permission for the queried datasets."],["AnnotationSetUpdateCall","Updates an annotation set. The update must respect all mutability restrictions and other invariants described on the annotation set resource. Caller must have WRITE permission for the associated dataset."],["AnnotationUpdateCall","Updates an annotation. The update must respect all mutability restrictions and other invariants described on the annotation resource. Caller must have WRITE permission for the associated dataset."],["BatchAnnotationsResponse","There is no detailed description."],["BatchAnnotationsResponseEntry","There is no detailed description."],["BatchAnnotationsResponseEntryStatus",""],["BatchCreateAnnotationsRequest","There is no detailed description."],["Call","A call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a call set with the name NA12345."],["CallReadGroupSetsRequest","The read group set call request."],["CallReadGroupSetsResponse","The read group set call response."],["CallSet","A call set is a collection of variant calls, typically for one sample. It belongs to a variant set."],["CallsetCreateCall","Creates a new call set."],["CallsetDeleteCall","Deletes a call set."],["CallsetGetCall","Gets a call set by ID."],["CallsetMethods","A builder providing access to all methods supported on *callset* resources. It is not used directly, but through the `Genomics` hub."],["CallsetPatchCall","Updates a call set. This method supports patch semantics."],["CallsetSearchCall","Gets a list of call sets matching the criteria."],["CallsetUpdateCall","Updates a call set."],["CigarUnit","A single CIGAR operation."],["CoverageBucket","A bucket over which read coverage has been precomputed. A bucket corresponds to a specific range of the reference sequence."],["Dataset","A Dataset is a collection of genomic data."],["DatasetCreateCall","Creates a new dataset."],["DatasetDeleteCall","Deletes a dataset."],["DatasetGetCall","Gets a dataset by ID."],["DatasetListCall","Lists datasets within a project."],["DatasetMethods","A builder providing access to all methods supported on *dataset* resources. It is not used directly, but through the `Genomics` hub."],["DatasetPatchCall","Updates a dataset. This method supports patch semantics."],["DatasetUndeleteCall","Undeletes a dataset by restoring a dataset which was deleted via this API. This operation is only possible for a week after the deletion occurred."],["DatasetUpdateCall","Updates a dataset."],["DefaultDelegate","A delegate with a conservative default implementation, which is used if no other delegate is set."],["ErrorResponse","A utility to represent detailed errors we might see in case there are BadRequests. The latter happen if the sent parameters or request structures are unsound"],["ExperimentalCreateJobRequest","The job creation request."],["ExperimentalCreateJobResponse","The job creation response."],["ExperimentalJobCreateCall","Creates and asynchronously runs an ad-hoc job. This is an experimental call and may be removed or changed at any time."],["ExperimentalMethods","A builder providing access to all methods supported on *experimental* resources. It is not used directly, but through the `Genomics` hub."],["ExportReadGroupSetsRequest","The read group set export request."],["ExportReadGroupSetsResponse","The read group set export response."],["ExportVariantSetRequest","The variant data export request."],["ExportVariantSetResponse","The variant data export response."],["ExternalId",""],["FastqMetadata","There is no detailed description."],["Genomics","Central instance to access all Genomics related resource activities"],["ImportReadGroupSetsRequest","The read group set import request."],["ImportReadGroupSetsResponse","The read group set import response."],["ImportVariantsRequest","The variant data import request."],["ImportVariantsResponse","The variant data import response."],["Int32Value","Wrapper message for int32."],["InterleavedFastqSource","Describes an interleaved FASTQ file source for alignment."],["Job","A Job represents an ongoing process that can be monitored for status information."],["JobCancelCall","Cancels a job by ID. Note that it is possible for partial results to be generated and stored for cancelled jobs."],["JobGetCall","Gets a job by ID."],["JobMethods","A builder providing access to all methods supported on *job* resources. It is not used directly, but through the `Genomics` hub."],["JobRequest","A summary representation of the service request that spawned the job."],["JobSearchCall","Gets a list of jobs matching the criteria."],["LinearAlignment","A linear alignment can be represented by one CIGAR string. Describes the mapped position and local alignment of the read to the reference."],["ListBasesResponse","There is no detailed description."],["ListCoverageBucketsResponse","There is no detailed description."],["ListDatasetsResponse","The dataset list response."],["MergeVariantsRequest","There is no detailed description."],["Metadata","Metadata describes a single piece of variant call metadata. These data include a top level key and either a single value string (value) or a list of key-value pairs (info.) Value and info are mutually exclusive."],["MethodInfo","Contains information about an API request."],["MultiPartReader","Provides a `Read` interface that converts multiple parts into the protocol identified by RFC2387. **Note**: This implementation is just as rich as it needs to be to perform uploads to google APIs, and might not be a fully-featured implementation."],["PairedFastqSource","Describes a paired-end FASTQ file source for alignment."],["Position","An abstraction for referring to a genomic position, in relation to some already known reference. For now, represents a genomic position as a reference name, a base number on that reference (0-based), and a determination of forward or reverse strand."],["QueryRange","A 0-based half-open genomic coordinate range for search requests."],["Range","A 0-based half-open genomic coordinate range over a reference sequence."],["RangePosition","A 0-based half-open genomic coordinate range over a reference sequence, for representing the position of a genomic resource."],["Read","A read alignment describes a linear alignment of a string of DNA to a reference sequence, in addition to metadata about the fragment (the molecule of DNA sequenced) and the read (the bases which were read by the sequencer). A read is equivalent to a line in a SAM file. A read belongs to exactly one read group and exactly one read group set. Generating a reference-aligned sequence string When interacting with mapped reads, it's often useful to produce a string representing the local alignment of the read to reference. The following pseudocode demonstrates one way of doing this: out = \"\" offset = 0 for c in read.alignment.cigar { switch c.operation { case \"ALIGNMENT_MATCH\", \"SEQUENCE_MATCH\", \"SEQUENCE_MISMATCH\": out += read.alignedSequence[offset:offset+c.operationLength] offset += c.operationLength break case \"CLIP_SOFT\", \"INSERT\": offset += c.operationLength break case \"PAD\": out += repeat(\"*\", c.operationLength) break case \"DELETE\": out += repeat(\"-\", c.operationLength) break case \"SKIP\": out += repeat(\" \", c.operationLength) break case \"CLIP_HARD\": break } } return out Converting to SAM's CIGAR string The following pseudocode generates a SAM CIGAR string from the cigar field. Note that this is a lossy conversion (cigar.referenceSequence is lost). cigarMap = { \"ALIGNMENT_MATCH\": \"M\", \"INSERT\": \"I\", \"DELETE\": \"D\", \"SKIP\": \"N\", \"CLIP_SOFT\": \"S\", \"CLIP_HARD\": \"H\", \"PAD\": \"P\", \"SEQUENCE_MATCH\": \"=\", \"SEQUENCE_MISMATCH\": \"X\", } cigarStr = \"\" for c in read.alignment.cigar { cigarStr += c.operationLength + cigarMap[c.operation] } return cigarStr"],["ReadGroup","A read group is all the data that's processed the same way by the sequencer."],["ReadGroupExperiment","There is no detailed description."],["ReadGroupProgram","There is no detailed description."],["ReadGroupSet","A read group set is a logical collection of read groups, which are collections of reads produced by a sequencer. A read group set typically models reads corresponding to one sample, sequenced one way, and aligned one way. - A read group set belongs to one dataset. - A read group belongs to one read group set. - A read belongs to one read group."],["ReadMethods","A builder providing access to all methods supported on *read* resources. It is not used directly, but through the `Genomics` hub."],["ReadSearchCall","Gets a list of reads for one or more read group sets. Reads search operates over a genomic coordinate space of reference sequence & position defined over the reference sequences to which the requested read group sets are aligned."],["ReadgroupsetAlignCall","Aligns read data from existing read group sets or files from Google Cloud Storage. See the  alignment and variant calling documentation for more details."],["ReadgroupsetCallCall","Calls variants on read data from existing read group sets or files from Google Cloud Storage. See the  alignment and variant calling documentation for more details."],["ReadgroupsetCoveragebucketListCall","Lists fixed width coverage buckets for a read group set, each of which correspond to a range of a reference sequence. Each bucket summarizes coverage information across its corresponding genomic range."],["ReadgroupsetDeleteCall","Deletes a read group set."],["ReadgroupsetExportCall","Exports read group sets to a BAM file in Google Cloud Storage."],["ReadgroupsetGetCall","Gets a read group set by ID."],["ReadgroupsetImportCall","Creates read group sets by asynchronously importing the provided information."],["ReadgroupsetMethods","A builder providing access to all methods supported on *readgroupset* resources. It is not used directly, but through the `Genomics` hub."],["ReadgroupsetPatchCall","Updates a read group set. This method supports patch semantics."],["ReadgroupsetSearchCall","Searches for read group sets matching the criteria."],["ReadgroupsetUpdateCall","Updates a read group set."],["Reference","A reference is a canonical assembled DNA sequence, intended to act as a reference coordinate space for other genomic annotations. A single reference might represent the human chromosome 1 or mitochandrial DNA, for instance. A reference belongs to one or more reference sets."],["ReferenceBaseListCall","Lists the bases in a reference, optionally restricted to a range."],["ReferenceBound","ReferenceBound records an upper bound for the starting coordinate of variants in a particular reference."],["ReferenceGetCall","Gets a reference."],["ReferenceMethods","A builder providing access to all methods supported on *reference* resources. It is not used directly, but through the `Genomics` hub."],["ReferenceSearchCall","Searches for references which match the given criteria."],["ReferenceSet","A reference set is a set of references which typically comprise a reference assembly for a species, such as GRCh38 which is representative of the human genome. A reference set defines a common coordinate space for comparing reference-aligned experimental data. A reference set contains 1 or more references."],["ReferencesetGetCall","Gets a reference set."],["ReferencesetMethods","A builder providing access to all methods supported on *referenceset* resources. It is not used directly, but through the `Genomics` hub."],["ReferencesetSearchCall","Searches for reference sets which match the given criteria."],["SearchAnnotationSetsRequest","There is no detailed description."],["SearchAnnotationSetsResponse","There is no detailed description."],["SearchAnnotationsRequest","There is no detailed description."],["SearchAnnotationsResponse","There is no detailed description."],["SearchCallSetsRequest","The call set search request."],["SearchCallSetsResponse","The call set search response."],["SearchJobsRequest","The jobs search request."],["SearchJobsResponse","The job search response."],["SearchReadGroupSetsRequest","The read group set search request."],["SearchReadGroupSetsResponse","The read group set search response."],["SearchReadsRequest","The read search request."],["SearchReadsResponse","The read search response."],["SearchReferenceSetsRequest",""],["SearchReferenceSetsResponse","There is no detailed description."],["SearchReferencesRequest",""],["SearchReferencesResponse","There is no detailed description."],["SearchVariantSetsRequest","The search variant sets request."],["SearchVariantSetsResponse","The search variant sets response."],["SearchVariantsRequest","The variant search request."],["SearchVariantsResponse","The variant search response."],["StreamReadsRequest","There is no detailed description."],["StreamReadsResponse","There is no detailed description."],["StreamingReadstoreMethods","A builder providing access to all methods supported on *streamingReadstore* resources. It is not used directly, but through the `Genomics` hub."],["StreamingReadstoreStreamreadCall","Gets a stream of reads for one or more read group sets. Reads search operates over a genomic coordinate space of reference sequence & position defined over the reference sequences to which the requested read group sets are aligned."],["Transcript","A transcript represents the assertion that a particular region of the reference genome may be transcribed as RNA."],["TranscriptCodingSequence","There is no detailed description."],["TranscriptExon","There is no detailed description."],["Variant","A variant represents a change in DNA sequence relative to a reference sequence. For example, a variant could represent a SNP or an insertion. Variants belong to a variant set. Each of the calls on a variant represent a determination of genotype with respect to that variant. For example, a call might assign probability of 0.32 to the occurrence of a SNP named rs1234 in a sample named NA12345. A call belongs to a call set, which contains related calls typically from one sample."],["VariantAnnotation","A Variant annotation."],["VariantAnnotationCondition",""],["VariantCreateCall","Creates a new variant."],["VariantDeleteCall","Deletes a variant."],["VariantGetCall","Gets a variant by ID."],["VariantMethods","A builder providing access to all methods supported on *variant* resources. It is not used directly, but through the `Genomics` hub."],["VariantSearchCall","Gets a list of variants matching the criteria."],["VariantSet","A variant set is a collection of call sets and variants. It contains summary statistics of those contents. A variant set belongs to a dataset."],["VariantUpdateCall","Updates a variant's names and info fields. All other modifications are silently ignored. Returns the modified variant without its calls."],["VariantsetDeleteCall","Deletes the contents of a variant set. The variant set object is not deleted."],["VariantsetExportCall","Exports variant set data to an external destination."],["VariantsetGetCall","Gets a variant set by ID."],["VariantsetImportVariantCall","Creates variant data by asynchronously importing the provided information."],["VariantsetMergeVariantCall","Merges the given variants with existing variants. Each variant will be merged with an existing variant that matches its reference sequence, start, end, reference bases, and alternative bases. If no such variant exists, a new one will be created."],["VariantsetMethods","A builder providing access to all methods supported on *variantset* resources. It is not used directly, but through the `Genomics` hub."],["VariantsetPatchCall","Updates a variant set's metadata. All other modifications are silently ignored. This method supports patch semantics."],["VariantsetSearchCall","Returns a list of all variant sets matching search criteria."],["VariantsetUpdateCall","Updates a variant set's metadata. All other modifications are silently ignored."]],"trait":[["CallBuilder","Identifies types which represent builders for a particular resource method"],["Delegate","A trait specifying functionality to help controlling any request performed by the API. The trait has a conservative default implementation."],["Hub","Identifies the Hub. There is only one per library, this trait is supposed to make intended use more explicit. The hub allows to access all resource methods more easily."],["MethodsBuilder","Identifies types for building methods of a particular resource type"],["NestedType","Identifies types which are only used by other types internally. They have no special meaning, this trait just marks them for completeness."],["Part","Identifies types which are only used as part of other types, which usually are carrying the `Resource` trait."],["ReadSeek","A utility to specify reader types which provide seeking capabilities too"],["RequestValue","Identifies types which are used in API requests."],["Resource","Identifies types which can be inserted and deleted. Types with this trait are most commonly used by clients of this API."],["ResponseResult","Identifies types which are used in API responses."],["ToParts","A trait for all types that can convert themselves into a *parts* string"]],"type":[["Result","A universal result type used as return for all calls."]]});